### PR TITLE
allow NNlib v0.9

### PR DESCRIPTION
**Pull request by CarloLucibello**
Also unbreakes a gpu test now passing on julia v1.9

### CODE REVIEW

File: test/gpu.jl [hunk 1]
```diff
@@ -23,7 +23,11 @@ end
   @test (repr("text/plain", y); true)
 
   gA = rand(3, 2) |> cu;
-  @test_broken gradient(A -> sum(A * y), gA)[1] isa CuArray  # fails with JLArray, bug in Zygote?
+  if VERSION >= v"1.9" && CUDA.functional()
```
Comment (ToucheSir) [inline test/gpu.jl:26]: Based on the comment below, shouldn't this be a check for if `gA isa JLArray`? Otherwise LGTM

Comment (CarloLucibello) [inline test/gpu.jl:26]:
```suggestion
  if VERSION >= v"1.9" && ga isa JLArray
```

Comment (CarloLucibello) [reply test/gpu.jl:26]: JLArray.jl is not loaded if cuda is functional

Comment (CarloLucibello) [reply test/gpu.jl:26]: All cpu/gpu tests are passing in any case. Merging as I need a new release to continue the work on the Flux cuda extension.

**Pull request merged**